lib/occ-helper.ts

```diff
@@ -1093,7 +1093,7 @@ export class OccHelper {
         ) {
             const edge = this.occ.TopoDS.Edge_1(anExplorer.Current());
             const edgeHash = edge.HashCode(100000000);
-            if (!edgeHashes.hasOwnProperty(edgeHash)) {
```

Comment:
yeah, this old code indeed is quite strange... I wasn't really sure about the details, I trust your solution will work fine here :)

Reply:
It needs a test. Do you know where an original example is, like in opencacase.js because it just doesn't seem right to me.

Reply:
Hey, I was looking at that code now with more attention and I get why this is needed.

For original reference I did use CascadeStudio for inspiration:
https://github.com/zalo/CascadeStudio/blob/master/js/CADWorker/CascadeStudioStandardLibrary.js

I'll try to explain.

edgeHashes is declared as an object above to just collect all the hashes of shapes as properties on the object that were already returned in order not to return the same shape (edge, a face or something else) multiple times. Practically one edge can be shared by two faces, so I suspect those iterations might return the same edge few times, but this is not a behaviour that is good in this case. Having hash properties on that object, like edgeHashes.123123141 is mostly a performance thing.

You could add already returned hashes to an array, but then you would need to do something like edgeHashes.find(...) and that is much more expensive computation then a simple check if the property on the object exists or not. If it does not exist, we call callback and add that property to the object.

I think that hasOwnProperty isn't really needed here and we could do something like this:
if(edgeHases[edgeHash] !== undefined) {

This would do the same trick.